server/client_manager.py

```diff
@@ -34,11 +34,14 @@ class Client:
 
         Clients may only belong to a single room.
         """
-
+        
         def __init__(self, server, transport: asyncio.Transport, user_id: int, ipid: int):
             self.is_checked = False
             self.transport = transport
             self.hdid = ''
+            self.release = ''
+            self.major_version = ''
+            self.minor_version = ''
```

Comment:
What type of versioning are you using? I think the meaning of these doesn't match.
https://semver.org/
If we're using semantic versioning, "release" doesn't make any sense. 

Right now, you have it as `release.major_version.minor_version`
When it should be `major_version.minor_version.patch_version`

Reply:
@in1tiate 
To add more context, here's the wikipedia article as well https://en.wikipedia.org/wiki/Software_versioning

> Other schemes impart meaning on individual sequences:
>    major.minor[.build[.revision]] (example: 1.2.12.102)
>   major.minor[.maintenance[.build]] (example: 1.4.3.5249)

Reply:
@oldmud0 you handle this one, all I did was copy what tsuserver's self-versioning does

Reply:
okay basically. @caleb-mabry. tsuserver and AO2 do not use semver. I know it looks like semver and mostly operates like semver but it isn't semver.

Reply:
If it's self versioning, that's fine!

Reply:
@caleb-mabry I'm aware of semver. Unfortunately the client does not follow [semver terminology either](https://github.com/AttorneyOnline/AO2-Client/blob/master/include/aoapplication.h#L511).

Reply:
This is just the first I've seen of it. So long as it's consistent between client and server it shouldn't be a problem. Thanks for the info